x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/UserBoolQueryBuilder.java

```diff
@@ -30,6 +33,13 @@ public class UserBoolQueryBuilder extends BoolQueryBuilder {
         List.of(exact("username"), exact("roles"), exact("full_name"), exact("email"), exact("enabled"))
     );
 
+    private static final List<String> ALLOWED_FIELD_NAMES = List.of(
+        RoutingFieldMapper.NAME,
+        IgnoredFieldMapper.NAME,
+        LegacyTypeFieldMapper.NAME,
+        "type"
+    );
```

Comment:
Can you describe the problem you encountered that made you add two new lists with the same metadata field names? I guess that the stored fields phase was previously skipping these checks that the fetch fields phase now bumps into?

Reply:
I might actually need to remve tese from the list after I introduced `fetchStoedFields` in `FieldFetcher`

Reply:
Anyway the reason behind it is this, I will double check if that is strictly required since after introducing `fetchStoredFields` and checking `ft == null` in `FieldFetcher` this might not be needed anymore (as a side effect).

```
// If the field is not allowed, behave as if it is not mapped
        if (allowedFields != null && false == allowedFields.test(name)) {
            return null;
        }
```

Some security related queries like UserBookQuery and ApiBoolQuery use this setting to allow/disallow fields.
The `MappedFieldType` is not available in that case which means a `ValueFecther` is not available.

Reply:
Honestly I find this way of handling allowed fields quite obscure and definitely not easy to get until you step on a bug or test failure.

Reply:
This was added in this pr [8c09fc8f9fa9e9c75521491594901fc3b3e6023c](https://github.com/elastic/elasticsearch/commit/8c09fc8f9fa9e9c75521491594901fc3b3e6023c)

Reply:
Maybe it would have been better to return a subclass of `MappedFieldType`...something like `SecurityHiddenMappedFieldType` which might return null/empty values from its `ValueFetcher` to hide details that are expected to be hidden?

Reply:
In summary this might not be required anymore because when `allowedFields` is used the `MappedFieldType` is null as explained above and we skip those fields in the `FieldFetcher`.